src/java/org/apache/cassandra/config/ParameterizedClass.java

```diff
@@ -74,32 +77,56 @@ static public <K> K newInstance(ParameterizedClass parameterizedClass, List<Stri
                 if (!searchPackage.isEmpty() && !searchPackage.endsWith("."))
                     searchPackage = searchPackage + '.';
                 String name = searchPackage + parameterizedClass.class_name;
-                Class<?> providerClass = Class.forName(name);
-                try
-                {
-                    Constructor<?> constructor = providerClass.getConstructor(Map.class);
-                    K instance = (K) constructor.newInstance(parameterizedClass.parameters);
-                    return instance;
-                }
-                catch (Exception constructorEx)
-                {
-                    //no-op
-                }
-                // fallback to no arg constructor if no params present
-                if (parameterizedClass.parameters == null || parameterizedClass.parameters.isEmpty())
-                {
-                    Constructor<?> constructor = providerClass.getConstructor();
-                    K instance = (K) constructor.newInstance();
-                    return instance;
-                }
+                providerClass = Class.forName(name);
             }
-            // there are about 5 checked exceptions that could be thrown here.
-            catch (Exception e)
+            catch (ClassNotFoundException e)
             {
-                last = e;
+                //no-op
             }
         }
-        throw new ConfigurationException("Unable to create parameterized class " + parameterizedClass.class_name, last);
+
+        if (providerClass == null)
+        {
+            String pkgList = '[' + searchPackages.stream().map(p -> '"' + p + '"').collect(Collectors.joining(",")) + ']';
+            String error = "Unable to find class " + parameterizedClass.class_name + " in packages " + pkgList;
+            throw new ConfigurationException(error);
+        }
+
+        try
+        {
+            Constructor<?>[] declaredConstructors = providerClass.getDeclaredConstructors();
+
+            Constructor mapConstructor = Arrays.stream(declaredConstructors)
```

Comment:
no need to use streams here. I think a simple helper method filtering on a predicate over declared constructors is just fine imo. There was also a thread on ML about avoiding streams when possible, even this is not a hot path and streams might be just fine, we can definitely go just without.

Reply:
@tiagomlalves yes, as in this PR https://issues.apache.org/jira/browse/CASSANDRA-20368?focusedCommentId=17933231&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-17933231

Reply:
fixed.